### ISSUE TITLE
Does not appear to work with Next 13 Server Components

### ISSUE THREAD

**Issue by rblackman** [portabletext/react-portabletext#27]
This package does not seem to be compatible with Next 13's [App directory/server components](https://beta.nextjs.org/docs/routing/fundamentals). When used with Next.js 13, it results in the following error:

> Server Error
TypeError: w.createContext is not a function

**Comment by jezpoz** [comment]
Hi @rblackman 

I tried this with Next.js 13 as well, and I got the same error. But then I read about server components vs. client components.
It worked if I created a PortableText component like this:

```
import { PortableText } from '@portabletext/react';
import type { PortableTextComponents } from '@portabletext/react';

const portabletextComponents: PortableTextComponents = {
  block: {
    h1: ({children}) => <h1 className="text-3xl">{children}</h1>,
    h2: ({children}) => <h2 className="text-xl">{children}</h2>,
    h3: ({children}) => <h3 className="text-lg">{children}</h3>,
    normal: ({children}) => <p className="text-base">{children}</p>
  }
}

export function Content({blocks}) {
  return (
    <PortableText value={blocks} components={portabletextComponents} />
  )
}
```

If you add the `'use client;'` at top of this file, it will render as a client component [described here](https://beta.nextjs.org/docs/rendering/server-and-client-components#convention)

**Comment by rblackman** [comment]
Hmm, I thought I tried that and got the same error. Rereading the documentation again, I think I may have put the `'use client'` line after my imports so I'm wondering if it didn't actually render as a client component. I will experiment more and see if I can come up with reliable reproduction steps.

I guess as a feature request, it would be nice if this worked with server components. I don't know how practical that request is.

**Comment by alvarlagerlof** [comment]
It worked for me with "use client" too, but it's not ideal, because then the server cannot render an initial HTM during SSR. Would also love if this component worked with server components too!

**Comment by pm0u** [comment]
Also ran into this same error, agree - workarounds are somewhat moot. Will have to wait for a fix. Has anyone tried Next 13/React 18 using the pages directory?

**Comment by alvarlagerlof** [comment]
> Has anyone tried Next 13/React 18 using the pages directory?

I have, my whole site is upgraded to it. https://alvar.dev and https://github.com/alvarlagerlof/portfolio. Server components can't do client context, which this library makes use of. So that's why. I had to add "use client" to my block content render component as described above. That causes it to need to load (render) on blog posts.

**Comment by rexxars** [comment]
We just released `@portabletext/react@2.0.0` which should support rendering through React Server Components. Do note that there are a few breaking changes in this release, if you were using context to pass down component configuration: https://github.com/portabletext/react-portabletext/releases/tag/v2.0.0
